include/oneapi/dpl/pstl/hetero/dpcpp/utils_ranges_sycl.h

```diff
@@ -517,9 +517,8 @@ struct __get_sycl_range
         auto buf = copy_direct(__first, __last, copy_direct_tag());
         buf = copy_back(__first, buf, copy_back_tag());
 
-        auto __p_buf = ::std::shared_ptr<oneapi::dpl::__internal::__lifetime_keeper<decltype(buf)>>(
-            new oneapi::dpl::__internal::__lifetime_keeper<decltype(buf)>(buf));
-        m_buffers.push_back(__p_buf);
+        using BufferType = oneapi::dpl::__internal::__lifetime_keeper<decltype(buf)>;
+        m_buffers.push_back(::std::make_shared<BufferType>(buf));
```

Comment:
Here we have only one memory allocation at whole ```::std::make_shared<BufferType>(buf)```
 - for details please see https://en.cppreference.com/w/cpp/memory/shared_ptr/make_shared
And call ```void push_back( T&& value )``` without extra ```addRef / releaseRef``` calls